doc/advanced/content/executor_design.rst

```diff
@@ -113,105 +113,101 @@ Typical examples of an execution context are a thread pool or a runtime environm
 **Execution Agent**
 
 It is a unit of execution of a specific execution context that is mapped to a single invocation
-of a callable function on an execution resource. An execution agent can too have different semantics which
-are derived from the execution context.
+of a callable function on an execution resource. An execution agent can have different semantics that
+is derived from the execution context.
```

Comment:
this should be plural. the semantics are derived (plural) not the execution agent (singular)